fdbclient/KeyRangeMap.actor.cpp

```diff
@@ -254,3 +308,87 @@ Future<Void> krmSetRangeCoalescing(Reference<ReadYourWritesTransaction> const& t
                                    Value const& value) {
 	return holdWhile(tr, krmSetRangeCoalescing_(tr.getPtr(), mapPrefix, range, maxRange, value));
 }
+
+TEST_CASE("/keyrangemap/decoderange/aligned") {
+	Arena arena;
+	Key prefix = LiteralStringRef("/prefix/");
+	StringRef fullKeyA = StringRef(arena, LiteralStringRef("/prefix/a"));
+	StringRef fullKeyB = StringRef(arena, LiteralStringRef("/prefix/b"));
+	StringRef fullKeyC = StringRef(arena, LiteralStringRef("/prefix/c"));
+	StringRef fullKeyD = StringRef(arena, LiteralStringRef("/prefix/d"));
+
+	StringRef keyA = StringRef(arena, LiteralStringRef("a"));
+	StringRef keyB = StringRef(arena, LiteralStringRef("b"));
+	StringRef keyC = StringRef(arena, LiteralStringRef("c"));
+	StringRef keyD = StringRef(arena, LiteralStringRef("d"));
+	StringRef keyE = StringRef(arena, LiteralStringRef("e"));
+	StringRef keyAB = StringRef(arena, LiteralStringRef("ab"));
+	StringRef keyCD = StringRef(arena, LiteralStringRef("cd"));
+
+	// Fake getRange() call.
+	RangeResult kv;
+	kv.push_back(arena, KeyValueRef(fullKeyA, keyA));
+	kv.push_back(arena, KeyValueRef(fullKeyB, keyB));
+	kv.push_back(arena, KeyValueRef(fullKeyC, keyC));
+	kv.push_back(arena, KeyValueRef(fullKeyD, keyD));
+
+	// [A, AB(start), B, C, CD(end), D]
+	RangeResult decodedRanges = krmDecodeRanges(prefix, KeyRangeRef(keyAB, keyCD), kv);
+	ASSERT(decodedRanges.size() == 4);
+	ASSERT(decodedRanges.front().key == keyAB);
```

Comment:
I thought the idea of aligned was that the start/end ranges in the result shouldn't be truncated to the query begin/end, so this part of the asserts in these 2 tests should be flipped?
If the test passes with this either something is wrong with my understanding, or with the code.

More generally it also might be good to add tests for the "off the end of the prefix" begin key/end key case?

Reply:
I think your understanding is flipped and maybe the naming is a bit unintuitive. Aligned = aligned to asked boundaries. Unaligned = underlying data boundaries. By default `aligned = true`.